frontend/src/compliance_reporting/ScheduleBContainer.js

```diff
@@ -167,77 +167,112 @@ class ScheduleBContainer extends Component {
     }
   }
 
-  componentWillReceiveProps (nextProps) {
-    const { grid } = this.state;
+  componentWillReceiveProps(nextProps) {
+    const {grid} = this.state;
 
-    if (!nextProps.scheduleState.scheduleB || !nextProps.scheduleState.scheduleB.records) {
-      return;
-    }
+    if (nextProps.snapshot) {
+      // just use the snapshot
+      const source = nextProps.snapshot.scheduleB;
 
-    if ((grid.length - 2) < nextProps.scheduleState.scheduleB.records.length) {
-      this._addRow(nextProps.scheduleState.scheduleB.records.length - (grid.length - 2));
-    }
+      if ((grid.length - 2) < source.records.length) {
```

Comment:
Can we add a check for source here? and to make sure we can retrieve length from source.records?

I tested this with a compliance report that doesn't have schedule B filled in and it's running into errors